privateSBOMExchange/src/scripts/Redaction.py

```diff
@@ -150,78 +241,74 @@ def main():
+    trees[tree_name] =sbom_tree
+    print(trees)
+
     
-    cpabe_setup_file = "privateSBOMExchange/src/lib/cpabe-setup.so"
-    cpabe_keygen_file = "privateSBOMExchange/src/lib/cpabe-keygen.so"
-    cpabe_enc_file = "privateSBOMExchange/src/lib/cpabe-enc.so"
-    cpabe_dec_file = "privateSBOMExchange/src/lib/cpabe-dec.so"
+
+    #using tree stucture, store sbom as tree in sqlalchamey db
+    t1=try_tree()
+    
+    store_SBOM_as_tree_in_db(sbom_tree,sbom_tree.get(b"name"))      
+    retrieved_tree=retrieve_sbom_as_tree_from_db(sbom_tree.get(b"name"))  
+    if compare_stored_and_retrieved_tree(sbom_tree,retrieved_tree):
+        print("The Trees are identical in structures and nodes")
+
+
+    cpabe_setup_file = "src/lib/cpabe-setup.so"
+    cpabe_keygen_file = "src/lib/cpabe-keygen.so"
+    cpabe_enc_file = "src/lib/cpabe-enc.so"
+    cpabe_dec_file = "sr    c/lib/cpabe-dec.so"
 
 
     #testing CP-ABE setup algorithim
     
     print("\n\nTesting CP-ABE setup ....")
     cpabe_setup_functions = CDLL(cpabe_setup_file)
-    print(type(cpabe_setup_functions))
     cpabe_setup_functions.main()
 
     #testing CP-ABE encrypt algorithim
-    
-    print("\n\nTesting CP-ABE encrypt ....")
-    cpabe_encrypt_functions = CDLL(cpabe_enc_file)
-    print(type(cpabe_encrypt_functions))
-    cpabe_encrypt_functions.main.restype = c_int
-    cpabe_encrypt_functions.main.argtypes = c_int,POINTER(c_char_p)
-
-    def make_args(cmd):
-        args = cmd.encode().split()
-        return (c_char_p * len(args))(*args)
-
-    args = make_args('3 pub_key flatten_sbom foo')
-    print(*args)
     print(os.getpid())
-    #backup the file before deletion
-    cpabe_encrypt_functions.main(len(args), args)
-
-   
-   
-    #cpabe-keygen -o sara_priv_key pub_key master_key test/policy
+ 
+    print("\n\nTesting CP-ABE encrypt ....")
+    encrypt_SBOM(flatten_SBOM_data, cpabe_enc_file)
 
     #testing CP-ABE key generation algorithim
     print("\n\nTesting CP-ABE key generation ....")
-    cpabe_keygen_functions = CDLL(cpabe_keygen_file)
-
-    cpabe_keygen_functions.main.restype = c_int
-    cpabe_keygen_functions.main.argtypes = c_int,POINTER(c_char_p)
-
-    args = make_args('5 -o priv_key pub_key master_key foo')
-    print(*args)
-    cpabe_keygen_functions.main(len(args), args)
-
+    generate_user_private_key(cpabe_keygen_file)
 
-   
     #testing CP-ABE decrypt algorithim
     print("\n\nTesting CP-ABE decrypt ....")
-    cpabe_decrypt_functions = CDLL(cpabe_dec_file)
 
-    cpabe_decrypt_functions.main.restype = c_int
-    cpabe_decrypt_functions.main.argtypes = c_int,POINTER(c_char_p)
     #cpabe-dec PUB_KEY PRIV_KEY FILE
-    args = make_args('3 pub_key sara_priv_key flatten_sbom.cpabe')
-    print(*args)
-    cpabe_decrypt_functions.main(len(args), args)
+    decrypt_SBOM_field(cpabe_dec_file,"dataLicense","pub_key","priv_key")
 
 
-   
+
+ 
+# Create a database engine
+engine = create_engine('sqlite:///dfsddff.db')  # Use SQLite for this example
```

Comment:
You probably would want to use a better name for this. I'm happy making this a ticket and pick it up in a separate PR